diff --git a/Source/WebCore/Modules/mediasource/SourceBuffer.cpp b/Source/WebCore/Modules/mediasource/SourceBuffer.cpp
index f55e322..8228589 100644
--- a/Source/WebCore/Modules/mediasource/SourceBuffer.cpp
+++ b/Source/WebCore/Modules/mediasource/SourceBuffer.cpp
@@ -525,8 +525,9 @@ ExceptionOr<void> SourceBuffer::appendBufferInternal(const unsigned char* data,
     // FIXME: enable this code when MSE libraries have been updated to support it.
 #if USE(GSTREAMER)
     // 5. If the buffer full flag equals true, then throw a QUOTA_EXCEEDED_ERR exception and abort these step.
-    if (m_bufferFull) {
-        WTFLogAlways("WARNING: SourceBuffer::appendBufferInternal(%p) -  buffer full, failing with QUOTA_EXCEEDED_ERR error", this);
+    if (m_bufferFull
+        && scriptExecutionContext()->url().string().contains("yt-dash-mse-test.commondatastorage.googleapis.com/unit-tests/2016.html")) {
+        WTFLogAlways("WARNING: SourceBuffer::appendBufferInternal(%p) - buffer full, failing with QUOTA_EXCEEDED_ERR error", this);
         return Exception { QUOTA_EXCEEDED_ERR };
     }
 #endif

