--- a/Source/WebCore/platform/mediastream/openwebrtc/RealtimeMediaSourceCenterOwr.cpp	2017-03-31 17:15:31.567813078 +0300
+++ b/Source/WebCore/platform/mediastream/openwebrtc/RealtimeMediaSourceCenterOwr.cpp	2017-03-31 17:16:54.284918346 +0300
@@ -114,8 +114,11 @@
 
 Vector<CaptureDevice> RealtimeMediaSourceCenterOwr::getMediaStreamDevices()
 {
-    notImplemented();
-    return Vector<CaptureDevice>();
+    static Vector<CaptureDevice> devices = {
+        { "audioin", CaptureDevice::DeviceType::Audio, "Audio Input", "A/V Capture" },
+        { "videoin", CaptureDevice::DeviceType::Video, "Video Camera", "A/V Capture" },
+    };
+    return devices;
 }
 
 void RealtimeMediaSourceCenterOwr::mediaSourcesAvailable(GList* sources)
