diff --git a/src/browser/wpe/test/ocdm_client.cpp b/src/browser/wpe/test/ocdm_client.cpp
index 32033d6..bb4f5fd 100644
--- a/src/browser/wpe/test/ocdm_client.cpp
+++ b/src/browser/wpe/test/ocdm_client.cpp
@@ -151,7 +151,11 @@ int main()
   unsigned char* puc_lic_response = (unsigned char*)lic_response.c_str();
   int cb_lic_response = lic_response.length();
 
-  const char test_challenge[3096] = {0xde, 0xad, 0xf0, 0x01};
+  char test_challenge[3096];
+  test_challenge[0] = 0xde;
+  test_challenge[1] = 0xad; 
+  test_challenge[2] = 0xf0;
+  test_challenge[3] = 0x01;
   int challenge_len = 0;
   unsigned char test_url[100];
   int url_length = 0;
