diff --git a/configure b/configure
index 9110f12..6421a66 100755
--- a/configure
+++ b/configure
@@ -1624,7 +1624,6 @@ infodir
 docdir
 oldincludedir
 includedir
-runstatedir
 localstatedir
 sharedstatedir
 sysconfdir
@@ -2142,7 +2141,6 @@ datadir='${datarootdir}'
 sysconfdir='${prefix}/etc'
 sharedstatedir='${prefix}/com'
 localstatedir='${prefix}/var'
-runstatedir='${localstatedir}/run'
 includedir='${prefix}/include'
 oldincludedir='/usr/include'
 docdir='${datarootdir}/doc/${PACKAGE_TARNAME}'
@@ -2395,15 +2393,6 @@ do
   | -silent | --silent | --silen | --sile | --sil)
     silent=yes ;;
 
-  -runstatedir | --runstatedir | --runstatedi | --runstated \
-  | --runstate | --runstat | --runsta | --runst | --runs \
-  | --run | --ru | --r)
-    ac_prev=runstatedir ;;
-  -runstatedir=* | --runstatedir=* | --runstatedi=* | --runstated=* \
-  | --runstate=* | --runstat=* | --runsta=* | --runst=* | --runs=* \
-  | --run=* | --ru=* | --r=*)
-    runstatedir=$ac_optarg ;;
-
   -sbindir | --sbindir | --sbindi | --sbind | --sbin | --sbi | --sb)
     ac_prev=sbindir ;;
   -sbindir=* | --sbindir=* | --sbindi=* | --sbind=* | --sbin=* \
@@ -2541,7 +2530,7 @@ fi
 for ac_var in	exec_prefix prefix bindir sbindir libexecdir datarootdir \
 		datadir sysconfdir sharedstatedir localstatedir includedir \
 		oldincludedir docdir infodir htmldir dvidir pdfdir psdir \
-		libdir localedir mandir runstatedir
+		libdir localedir mandir
 do
   eval ac_val=\$$ac_var
   # Remove trailing slashes.
@@ -2694,7 +2683,6 @@ Fine tuning of the installation directories:
   --sysconfdir=DIR        read-only single-machine data [PREFIX/etc]
   --sharedstatedir=DIR    modifiable architecture-independent data [PREFIX/com]
   --localstatedir=DIR     modifiable single-machine data [PREFIX/var]
-  --runstatedir=DIR       modifiable per-process data [LOCALSTATEDIR/run]
   --libdir=DIR            object code libraries [EPREFIX/lib]
   --includedir=DIR        C header files [PREFIX/include]
   --oldincludedir=DIR     C header files for non-gcc [/usr/include]
@@ -34105,17 +34093,17 @@ fi
     old_LIBS=$LIBS
     old_CFLAGS=$CFLAGS
 
-        ac_fn_c_check_header_mongrel "$LINENO" "EGL/fbdev_window.h" "ac_cv_header_EGL_fbdev_window_h" "$ac_includes_default"
-if test "x$ac_cv_header_EGL_fbdev_window_h" = xyes; then :
+        ac_fn_c_check_header_mongrel "$LINENO" "EGL/eglplatform.h" "ac_cv_header_EGL_eglplatform_h" "$ac_includes_default"
+if test "x$ac_cv_header_EGL_eglplatform_h" = xyes; then :
 
-        LIBS="$LIBS -lUMP"
-        { $as_echo "$as_me:${as_lineno-$LINENO}: checking for mali_image_create in -lMali" >&5
-$as_echo_n "checking for mali_image_create in -lMali... " >&6; }
-if ${ac_cv_lib_Mali_mali_image_create+:} false; then :
+        LIBS="$LIBS "
+        { $as_echo "$as_me:${as_lineno-$LINENO}: checking for mali_egl_image_get_format in -lmali" >&5
+$as_echo_n "checking for mali_egl_image_get_format in -lmali... " >&6; }
+if ${ac_cv_lib_mali_mali_egl_image_get_format+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lMali  $LIBS"
+LIBS="-lmali  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -34125,29 +34113,29 @@ cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 #ifdef __cplusplus
 extern "C"
 #endif
-char mali_image_create ();
+char mali_egl_image_get_format ();
 int
 main ()
 {
-return mali_image_create ();
+return mali_egl_image_get_format ();
   ;
   return 0;
 }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"; then :
-  ac_cv_lib_Mali_mali_image_create=yes
+  ac_cv_lib_mali_mali_egl_image_get_format=yes
 else
-  ac_cv_lib_Mali_mali_image_create=no
+  ac_cv_lib_mali_mali_egl_image_get_format=no
 fi
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_Mali_mali_image_create" >&5
-$as_echo "$ac_cv_lib_Mali_mali_image_create" >&6; }
-if test "x$ac_cv_lib_Mali_mali_image_create" = xyes; then :
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_mali_mali_egl_image_get_format" >&5
+$as_echo "$ac_cv_lib_mali_mali_egl_image_get_format" >&6; }
+if test "x$ac_cv_lib_mali_mali_egl_image_get_format" = xyes; then :
 
-            LIBS="$LIBS -lMali"
+            LIBS="$LIBS"
             { $as_echo "$as_me:${as_lineno-$LINENO}: checking for glEnable in -lGLESv2" >&5
 $as_echo_n "checking for glEnable in -lGLESv2... " >&6; }
 if ${ac_cv_lib_GLESv2_glEnable+:} false; then :
@@ -34232,7 +34220,7 @@ if test "x$ac_cv_header_EGL_egl_h" = xyes; then :
 
                             HAVE_EGL=yes
                             HAVE_GLES2=yes
-                            EGL_LIBS="-lMali -lUMP"
+                            EGL_LIBS="-lmali "
                             EGL_CFLAGS=""
 
 $as_echo "#define USE_EGL_MALI_FB 1" >>confdefs.h
@@ -34429,12 +34417,12 @@ if test -n "$PKG_CONFIG"; then
         pkg_cv_WAYLAND_EGL_CFLAGS="$WAYLAND_EGL_CFLAGS"
     else
         if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"wayland-client >= 1.0 wayland-cursor >= 1.0 wayland-egl >= 9.0\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "wayland-client >= 1.0 wayland-cursor >= 1.0 wayland-egl >= 9.0") 2>&5
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"wayland-client >= 1.0 wayland-cursor >= 1.0 wayland-egl >= 1.0\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "wayland-client >= 1.0 wayland-cursor >= 1.0 wayland-egl >= 1.0") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_WAYLAND_EGL_CFLAGS=`$PKG_CONFIG --cflags "wayland-client >= 1.0 wayland-cursor >= 1.0 wayland-egl >= 9.0" 2>/dev/null`
+  pkg_cv_WAYLAND_EGL_CFLAGS=`$PKG_CONFIG --cflags "wayland-client >= 1.0 wayland-cursor >= 1.0 wayland-egl >= 1.0" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -34447,12 +34435,12 @@ if test -n "$PKG_CONFIG"; then
         pkg_cv_WAYLAND_EGL_LIBS="$WAYLAND_EGL_LIBS"
     else
         if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"wayland-client >= 1.0 wayland-cursor >= 1.0 wayland-egl >= 9.0\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "wayland-client >= 1.0 wayland-cursor >= 1.0 wayland-egl >= 9.0") 2>&5
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"wayland-client >= 1.0 wayland-cursor >= 1.0 wayland-egl >= 1.0\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "wayland-client >= 1.0 wayland-cursor >= 1.0 wayland-egl >= 1.0") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_WAYLAND_EGL_LIBS=`$PKG_CONFIG --libs "wayland-client >= 1.0 wayland-cursor >= 1.0 wayland-egl >= 9.0" 2>/dev/null`
+  pkg_cv_WAYLAND_EGL_LIBS=`$PKG_CONFIG --libs "wayland-client >= 1.0 wayland-cursor >= 1.0 wayland-egl >= 1.0" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -34471,9 +34459,9 @@ else
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        WAYLAND_EGL_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "wayland-client >= 1.0 wayland-cursor >= 1.0 wayland-egl >= 9.0"`
+	        WAYLAND_EGL_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "wayland-client >= 1.0 wayland-cursor >= 1.0 wayland-egl >= 1.0"`
         else
-	        WAYLAND_EGL_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "wayland-client >= 1.0 wayland-cursor >= 1.0 wayland-egl >= 9.0"`
+	        WAYLAND_EGL_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "wayland-client >= 1.0 wayland-cursor >= 1.0 wayland-egl >= 1.0"`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$WAYLAND_EGL_PKG_ERRORS" >&5
