From 6b26ebed0f21d255b93e76b73d400284354e4c94 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Enrique=20Oca=C3=B1a=20Gonz=C3=A1lez?= <eocanha@igalia.com>
Date: Tue, 17 Jan 2017 18:43:24 +0000
Subject: [PATCH 2/4] Fix decoder flushing

---
 omx/gstomx.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/omx/gstomx.c b/omx/gstomx.c
index 0237f3c..45732c3 100644
--- a/omx/gstomx.c
+++ b/omx/gstomx.c
@@ -1538,8 +1538,8 @@ gst_omx_port_set_flushing (GstOMXPort * port, GstClockTime timeout,
     last_error = OMX_ErrorNone;
     gst_omx_component_handle_messages (comp);
     while (signalled && last_error == OMX_ErrorNone && !port->flushed
-        && port->buffers
-        && port->buffers->len > g_queue_get_length (&port->pending_buffers)) {
+     /* && port->buffers
+        && port->buffers->len > g_queue_get_length (&port->pending_buffers) */) {
       signalled = gst_omx_component_wait_message (comp, timeout);
       if (signalled)
         gst_omx_component_handle_messages (comp);
-- 
1.8.3.2

