diff -Naur CURRENT/nexus/examples/hdmi_input/hdmi_input_hdcp_keyloader.c PATCHED/nexus/examples/hdmi_input/hdmi_input_hdcp_keyloader.c
--- CURRENT/nexus/examples/hdmi_input/hdmi_input_hdcp_keyloader.c	2015-10-08 08:24:42.757443000 +0000
+++ PATCHED/nexus/examples/hdmi_input/hdmi_input_hdcp_keyloader.c	2015-10-08 08:24:43.874450845 +0000
@@ -269,8 +269,8 @@
 int main(int argc, char **argv)
 {
     BSTD_UNUSED(argc);
-	printf("%d Platform has %d HDMI Inputs ; App requires one HDMI Input \n",
-		NEXUS_PLATFORM, NEXUS_NUM_HDMI_INPUTS);
+	printf("Platform has %d HDMI Inputs ; App requires one HDMI Input \n",
+		NEXUS_NUM_HDMI_INPUTS);
 
     printf("%s not supported on the %d platform.\n", argv[0], BCHP_CHIP) ;
     return 0 ;
diff -Naur CURRENT/nexus/examples/hdmi_input/hdmi_input_to_hdmi_output.c PATCHED/nexus/examples/hdmi_input/hdmi_input_to_hdmi_output.c
--- CURRENT/nexus/examples/hdmi_input/hdmi_input_to_hdmi_output.c	2015-10-08 08:24:42.757443000 +0000
+++ PATCHED/nexus/examples/hdmi_input/hdmi_input_to_hdmi_output.c	2015-10-08 08:24:43.875450852 +0000
@@ -1193,8 +1193,8 @@
 int main(int argc, char **argv)
 {
     BSTD_UNUSED(argc);
-    printf("%d Platform has %d HDMI Inputs and %d HDMI Outputs; App requires one of each.\n",
-        NEXUS_PLATFORM, NEXUS_NUM_HDMI_INPUTS, NEXUS_NUM_HDMI_OUTPUTS);
+    printf("Platform has %d HDMI Inputs and %d HDMI Outputs; App requires one of each.\n",
+        NEXUS_NUM_HDMI_INPUTS, NEXUS_NUM_HDMI_OUTPUTS);
 
     printf("%s not supported on the %d platform.\n", argv[0], BCHP_CHIP) ;
     return 0 ;
diff -Naur CURRENT/nexus/examples/hdmi_input/hdmi_repeater_passthrough.c PATCHED/nexus/examples/hdmi_input/hdmi_repeater_passthrough.c
--- CURRENT/nexus/examples/hdmi_input/hdmi_repeater_passthrough.c	2015-10-08 08:24:42.757443000 +0000
+++ PATCHED/nexus/examples/hdmi_input/hdmi_repeater_passthrough.c	2015-10-08 08:24:43.875450852 +0000
@@ -956,7 +956,7 @@
 int main(int argc, char **argv)
 {
     BSTD_UNUSED(argc);
-	printf("%d Platform has %d HDMI Inputs and %d HDMI Outputs; App requires one of each.\n",NEXUS_PLATFORM,
+	printf("Platform has %d HDMI Inputs and %d HDMI Outputs; App requires one of each.\n",
            NEXUS_NUM_HDMI_INPUTS, NEXUS_NUM_HDMI_OUTPUTS);
 
     printf("%s not supported on the %d platform.\n", argv[0], BCHP_CHIP) ;
