diff -Naur reference_code/nexus/modules/security/rawcommand_ext/include/nexus_security_rawcommand.h nexus/modules/security/rawcommand_ext/include/nexus_security_rawcommand.h
--- reference_code/nexus/modules/security/rawcommand_ext/include/nexus_security_rawcommand.h	1970-01-01 01:00:00.000000000 +0100
+++ nexus/modules/security/rawcommand_ext/include/nexus_security_rawcommand.h	2015-02-05 18:23:47.193572000 +0100
@@ -0,0 +1,81 @@
+/***************************************************************************
+*     (c)2004-2009 Broadcom Corporation
+*
+*  This program is the proprietary software of Broadcom Corporation and/or its licensors,
+*  and may only be used, duplicated, modified or distributed pursuant to the terms and
+*  conditions of a separate, written license agreement executed between you and Broadcom
+*  (an "Authorized License").  Except as set forth in an Authorized License, Broadcom grants
+*  no license (express or implied), right to use, or waiver of any kind with respect to the
+*  Software, and Broadcom expressly reserves all rights in and to the Software and all
+*  intellectual property rights therein.  IF YOU HAVE NO AUTHORIZED LICENSE, THEN YOU
+*  HAVE NO RIGHT TO USE THIS SOFTWARE IN ANY WAY, AND SHOULD IMMEDIATELY
+*  NOTIFY BROADCOM AND DISCONTINUE ALL USE OF THE SOFTWARE.
+*
+*  Except as expressly set forth in the Authorized License,
+*
+*  1.     This program, including its structure, sequence and organization, constitutes the valuable trade
+*  secrets of Broadcom, and you shall use all reasonable efforts to protect the confidentiality thereof,
+*  and to use this information only in connection with your use of Broadcom integrated circuit products.
+*
+*  2.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, THE SOFTWARE IS PROVIDED "AS IS"
+*  AND WITH ALL FAULTS AND BROADCOM MAKES NO PROMISES, REPRESENTATIONS OR
+*  WARRANTIES, EITHER EXPRESS, IMPLIED, STATUTORY, OR OTHERWISE, WITH RESPECT TO
+*  THE SOFTWARE.  BROADCOM SPECIFICALLY DISCLAIMS ANY AND ALL IMPLIED WARRANTIES
+*  OF TITLE, MERCHANTABILITY, NONINFRINGEMENT, FITNESS FOR A PARTICULAR PURPOSE,
+*  LACK OF VIRUSES, ACCURACY OR COMPLETENESS, QUIET ENJOYMENT, QUIET POSSESSION
+*  OR CORRESPONDENCE TO DESCRIPTION. YOU ASSUME THE ENTIRE RISK ARISING OUT OF
+*  USE OR PERFORMANCE OF THE SOFTWARE.
+*
+*  3.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, IN NO EVENT SHALL BROADCOM OR ITS
+*  LICENSORS BE LIABLE FOR (i) CONSEQUENTIAL, INCIDENTAL, SPECIAL, INDIRECT, OR
+*  EXEMPLARY DAMAGES WHATSOEVER ARISING OUT OF OR IN ANY WAY RELATING TO YOUR
+*  USE OF OR INABILITY TO USE THE SOFTWARE EVEN IF BROADCOM HAS BEEN ADVISED OF
+*  THE POSSIBILITY OF SUCH DAMAGES; OR (ii) ANY AMOUNT IN EXCESS OF THE AMOUNT
+*  ACTUALLY PAID FOR THE SOFTWARE ITSELF OR U.S. $1, WHICHEVER IS GREATER. THESE
+*  LIMITATIONS SHALL APPLY NOTWITHSTANDING ANY FAILURE OF ESSENTIAL PURPOSE OF
+*  ANY LIMITED REMEDY.
+*
+* $brcm_Workfile: nexus_security_rawcommand.h $
+* $brcm_Revision: 1 $
+* $brcm_Date: 10/1/09 1:22p $
+*
+* API Description:
+*   Management of timebase (clock rate) blocks.
+*
+* Revision History:
+*
+* $brcm_Log: /nexus/modules/security/7400/rawcommand_ext/include/nexus_security_rawcommand.h $
+* 
+* 1   10/1/09 1:22p erickson
+* SW7340-42: added new file
+*
+***************************************************************************/
+#ifndef NEXUS_SECURITY_RAWCOMMAND_H__
+#define NEXUS_SECURITY_RAWCOMMAND_H__
+
+#include "nexus_types.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*****************************************************************************
+Summary:
+Submit RAW BSP commands.
+
+Description
+This function is ONLY meant to be used by advanced users.
+*****************************************************************************/
+NEXUS_Error NEXUS_Security_SubmitRawCommand (
+    const unsigned int *pInBuffer,    /* attr{nelem=sizeInBuffer;reserved=90} */
+    unsigned int sizeInBuffer,        /* size of pInBuffer in integers */
+    unsigned int *pOutBuffer,         /* [out] attr{nelem=sizeOutBuffer;reserved=90} */
+    unsigned int sizeOutBuffer,        /* size of pOutBuffer in integers */
+    unsigned int *pActualOutSize      /* [out] number of integers written into pOutBuffer */
+    );
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif
diff -Naur reference_code/nexus/modules/security/rawcommand_ext/nexus_security_rawcommand.inc nexus/modules/security/rawcommand_ext/nexus_security_rawcommand.inc
--- reference_code/nexus/modules/security/rawcommand_ext/nexus_security_rawcommand.inc	1970-01-01 01:00:00.000000000 +0100
+++ nexus/modules/security/rawcommand_ext/nexus_security_rawcommand.inc	2015-02-05 18:23:47.194571000 +0100
@@ -0,0 +1,81 @@
+#############################################################################
+#    (c)2009-2012 Broadcom Corporation
+# 
+# This program is the proprietary software of Broadcom Corporation and/or its licensors,
+# and may only be used, duplicated, modified or distributed pursuant to the terms and
+# conditions of a separate, written license agreement executed between you and Broadcom
+# (an "Authorized License").  Except as set forth in an Authorized License, Broadcom grants
+# no license (express or implied), right to use, or waiver of any kind with respect to the
+# Software, and Broadcom expressly reserves all rights in and to the Software and all
+# intellectual property rights therein.  IF YOU HAVE NO AUTHORIZED LICENSE, THEN YOU
+# HAVE NO RIGHT TO USE THIS SOFTWARE IN ANY WAY, AND SHOULD IMMEDIATELY
+# NOTIFY BROADCOM AND DISCONTINUE ALL USE OF THE SOFTWARE.  
+#  
+# Except as expressly set forth in the Authorized License,
+#  
+# 1.     This program, including its structure, sequence and organization, constitutes the valuable trade
+# secrets of Broadcom, and you shall use all reasonable efforts to protect the confidentiality thereof,
+# and to use this information only in connection with your use of Broadcom integrated circuit products.
+#  
+# 2.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, THE SOFTWARE IS PROVIDED "AS IS" 
+# AND WITH ALL FAULTS AND BROADCOM MAKES NO PROMISES, REPRESENTATIONS OR 
+# WARRANTIES, EITHER EXPRESS, IMPLIED, STATUTORY, OR OTHERWISE, WITH RESPECT TO 
+# THE SOFTWARE.  BROADCOM SPECIFICALLY DISCLAIMS ANY AND ALL IMPLIED WARRANTIES 
+# OF TITLE, MERCHANTABILITY, NONINFRINGEMENT, FITNESS FOR A PARTICULAR PURPOSE, 
+# LACK OF VIRUSES, ACCURACY OR COMPLETENESS, QUIET ENJOYMENT, QUIET POSSESSION 
+# OR CORRESPONDENCE TO DESCRIPTION. YOU ASSUME THE ENTIRE RISK ARISING OUT OF 
+# USE OR PERFORMANCE OF THE SOFTWARE.
+# 
+# 3.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, IN NO EVENT SHALL BROADCOM OR ITS 
+# LICENSORS BE LIABLE FOR (i) CONSEQUENTIAL, INCIDENTAL, SPECIAL, INDIRECT, OR 
+# EXEMPLARY DAMAGES WHATSOEVER ARISING OUT OF OR IN ANY WAY RELATING TO YOUR 
+# USE OF OR INABILITY TO USE THE SOFTWARE EVEN IF BROADCOM HAS BEEN ADVISED OF 
+# THE POSSIBILITY OF SUCH DAMAGES; OR (ii) ANY AMOUNT IN EXCESS OF THE AMOUNT 
+# ACTUALLY PAID FOR THE SOFTWARE ITSELF OR U.S. $1, WHICHEVER IS GREATER. THESE 
+# LIMITATIONS SHALL APPLY NOTWITHSTANDING ANY FAILURE OF ESSENTIAL PURPOSE OF 
+# ANY LIMITED REMEDY.
+#
+# $brcm_Workfile: nexus_security_rawcommand.inc $
+# $brcm_Revision: 4 $
+# $brcm_Date: 7/20/12 12:27p $
+# 
+# Module Description:
+# 
+# Revision History:
+# 
+# $brcm_Log: /nexus/modules/security/rawcommand_ext/nexus_security_rawcommand.inc $
+# 
+# 4   7/20/12 12:27p erickson
+# SW7435-235: refactor directories to remove symbolic links
+# 
+# 3   9/23/11 6:33p atruong
+# SW7346-495: HSM_SOURCE_AVAILABLE option fix
+# 
+# 2   10/1/09 1:45p yili
+# SW7340-42:Folder rename
+# 
+# 1   9/30/09 3:14p yili
+# SW7340-42:Add raw command interface
+# 
+#############################################################################
+
+###################
+# Module Includes #
+###################
+NEXUS_SECURITY_PUBLIC_INCLUDES += $(NEXUS_TOP)/modules/security/rawcommand_ext/include
+
+##################
+# Module Sources #
+##################
+# if HSM_SOURCE_AVAILABLE=n is set, you must set NEXUS_EXTRALIBS to
+# include the pre-built hsm binary for this extension
+ifneq ($(HSM_SOURCE_AVAILABLE),n)
+NEXUS_SECURITY_SOURCES += \
+    $(NEXUS_TOP)/modules/security/rawcommand_ext/src/nexus_security_rawcommand.c
+endif
+
+##################
+# Module Defines #
+##################
+NEXUS_SECURITY_DEFINES += NEXUS_SECURITY_RAWCMD=1
+NEXUS_CFLAGS +=  -DBHSM_AUTO_TEST
